sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubMessageToRow.java

```diff
@@ -68,8 +69,21 @@
 
   public abstract boolean useDlq();
 
+  public abstract boolean useFlatSchema();
+
   private Schema payloadSchema() {
-    return messageSchema().getField(PAYLOAD_FIELD).getType().getRowSchema();
+    if (useFlatSchema()) {
+      Schema.Builder builder = Schema.builder();
+      for (Schema.Field field : messageSchema().getFields()) {
+        if (field.getName().equals(TIMESTAMP_FIELD)) {
+          continue;
+        }
+        builder.addField(field);
+      }
+      return builder.build();
+    } else {
+      return messageSchema().getField(PAYLOAD_FIELD).getType().getRowSchema();
```

Comment:
nit: shorter branch first is slightly easier to read

Reply:
Done